code/common/merge.q

```diff
@@ -41,7 +42,11 @@ getpartchunks:{[partdirs;mergelimit]
   /-get list of limits (rowcount or bytesize) to be used to get chunks of partitions to get merged in batch
   r:$[.merge.mergebybytelimit;exec bytes from t;exec rowcount from t];
   /-return list of partitions to be called in batch by merge by part function
-  (where r={$[z<x+y;y;x+y]}\[0;r;mergelimit]) cut exec ptdir from t
+  l:(where r={$[z<x+y;y;x+y]}\[0;r;mergelimit]),(select count i from .merge.partsizes)[`x];
+  /-set a limit to the number of partitions that can be in a chunk
+  s:-1_ asc raze {$[(l[x]-l[x-1])<=.merge.partlimit;l[x];l[x], l[x-1] + (v where 0=(v:1+ til l[x] - l[x-1]) mod .merge.partlimit)]} each til count l;
```

Comment:
does this work? it looks like `l` is referenced inside a lambda without being passed in, which I don't think will work?